typescript/sdk/src/token/checker.ts

```diff
@@ -25,6 +25,10 @@ export class HypERC20Checker extends HyperlaneRouterChecker<
   async checkChain(chain: ChainName): Promise<void> {
     await super.checkChain(chain);
     await this.checkToken(chain);
+    // We have adapted thi method to accept a proxyAdmin contract address parameter
+    // extending HypERC20Checker class with ProxiedRouterChecker, which could be done to avoid this change
+    // leads further issues as the ProxyAdmin contract does not conform to GasRouter interface
+    await this.checkProxiedContracts(chain, this.configMap[chain].proxyAdmin);
```

Comment:
I dont really understand this
> ProxyAdmin contract does not conform to GasRouter interface

Reply:
See my latest commit, it's my rough attempt to have `HypERC20Checker` extend `ProxiedRouterChecker`.
 
```
src/token/deploy.ts(200,9): error TS2322: Type 'TimelockController | ProxyAdmin | HypERC20 | HypERC20Collateral | HypXERC20Lockbox | HypXERC20 | ... 5 more ... | HypNativeScaled' is not assignable to type 'GasRouter'.
  Type 'TimelockController' is missing the following properties from type 'GasRouter': destinationGas, "destinationGas(uint32)", domains, "domains()", and 34 more.

15:39:19 - Found 1 error. Watching for file changes.
```

I don't think it make sense to add the `proxiedFactories` to `HypERC20Factories` anyways

Reply:
ah I see
yes agreed we should not add `proxiedFactories` to `HypERC20Factories`
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/1ec61debd2425785ca4ade111ad3cdaba8ccc8ac/typescript/sdk/src/router/ProxiedRouterDeployer.ts#L25-L35
here is what I did in the deployer

Reply:
This approach still leads to 

```
src/token/deploy.ts(200,9): error TS2322: Type 'TimelockController | ProxyAdmin | HypERC20 | HypERC20Collateral | HypXERC20Lockbox | HypXERC20 | ... 5 more ... | HypNativeScaled' is not assignable to type 'GasRouter'.
  Type 'TimelockController' is missing the following properties from type 'GasRouter': destinationGas, "destinationGas(uint32)", domains, "domains()", and 34 more.

15:39:19 - Found 1 error. Watching for file changes.
```

I think having HypERC20Checker extend ProxiedRouterChecker will be challenging